tests/queries/0_stateless/02293_compatibility_ignore_auto_increment_in_create_table.sql

```diff
@@ -0,0 +1,36 @@
+select 'disable AUTO_INCREMENT compatibility mode';
+set compatibility_ignore_auto_increment_in_create_table=false;
+
+select 'create table with AUTO_INCREMENT, compatibility disabled';
+DROP TABLE IF EXISTS ignore_auto_increment SYNC;
+CREATE TABLE ignore_auto_increment (
+    id int AUTO_INCREMENT
+) ENGINE=MergeTree() ORDER BY tuple(); -- {serverError SYNTAX_ERROR}
```

Comment:
Maybe add case with `AUTO_INCREMENT` and/or `DEFAULT` and/or `EPHEMERAL` for a single column.